crates/proof/proof-interop/src/consolidation.rs

```diff
@@ -205,6 +215,41 @@ where
 
         Ok(())
     }
+
+    /// Forms the replacement transaction inserted into a deposit-only block in the event that a
+    /// block is reduced due to invalid messages.
+    ///
+    /// <https://specs.optimism.io/interop/derivation.html#optimistic-block-deposited-transaction>
+    fn craft_replacement_transaction(old_header: &Sealed<Header>, old_output_root: B256) -> Bytes {
+        const REPLACEMENT_SENDER: Address = address!("deaddeaddeaddeaddeaddeaddeaddeaddead0002");
+        const REPLACEMENT_GAS: u64 = 36000;
+
+        let source = InteropBlockReplacementDepositSource::new(old_output_root);
+        let output_root_preimage = {
+            let mut raw_output = [0u8; 128];
+            raw_output[31] = 0x0;
+            raw_output[32..64].copy_from_slice(old_header.state_root.as_ref());
+            raw_output[64..96]
+                .copy_from_slice(old_header.withdrawals_root.unwrap_or_default().as_ref());
+            raw_output[96..128].copy_from_slice(old_header.hash().as_ref());
+            raw_output
+        };
```

Comment:
Will follow up in another PR with a dedicated `OutputRoot` type here that can handle encoding in the background. We do this enough that manual encoding is lazy and ugly. https://github.com/op-rs/kona/issues/1541

Reply:
#1544